modules/pipeline/providers/cms/pipeline_cms.go

```diff
@@ -251,26 +254,39 @@ func (c *pipelineCm) GetConfigs(ctx context.Context, ns string, globalDecrypt bo
 		}
 
 		// 配置项级别的展示配置
-		result[config.Key] = apistructs.PipelineCmsConfigValue{
+		result[config.Key] = &pb.PipelineCmsConfigValue{
 			Value:       vv,
 			EncryptInDB: *config.Encrypt,
 			Type:        config.Type,
 			Operations:  config.Extra.Operations,
 			Comment:     config.Extra.Comment,
 			From:        config.Extra.From,
-			TimeCreated: config.TimeCreated,
-			TimeUpdated: config.TimeUpdated,
+			TimeCreated: getPbTimestamp(config.TimeCreated),
+			TimeUpdated: getPbTimestamp(config.TimeUpdated),
 		}
 	}
 	return result, nil
 }
 
+func getPbTimestamp(t *time.Time) *timestamppb.Timestamp {
+	if t == nil {
+		return nil
+	}
+	return timestamppb.New(*t)
+}
+
 func getPipelineSourceFromContext(ctx context.Context) (apistructs.PipelineSource, error) {
-	pipelineSource, ok := ctx.Value(CtxKeyPipelineSource).(apistructs.PipelineSource)
-	if !ok || pipelineSource == "" {
+	var source apistructs.PipelineSource
+	switch ctx.Value(CtxKeyPipelineSource).(type) {
+	case apistructs.PipelineSource:
+		source = ctx.Value(CtxKeyPipelineSource).(apistructs.PipelineSource)
+	case string:
+		source = apistructs.PipelineSource(ctx.Value(CtxKeyPipelineSource).(string))
```

Comment:
what if neither apistructs.PipelineSource nor string? if imposible, you should comment reason in code

Reply:
add default branch.